src/client/components/MultiSelect/MultiSelect.tsx

```diff
@@ -1,91 +1,96 @@
+  const multiselectRef = useRef<HTMLDivElement>()
+  const [open, setOpen] = useState<boolean>(false)
+  const { t } = useTranslation()
+  const isValuesEmpty = Objects.isEmpty(values)
 
-  constructor(props: Props) {
-    super(props)
-    this.state = { open: false }
-    this.outsideClick = outsideClick(this)
-    window.addEventListener('click', this.outsideClick)
+  const localizeOption = (option: string) => {
+    const prefix = localizationPrefix ? `${localizationPrefix}.` : ''
+    return t(`${prefix}${option}`)
   }
 
-  componentWillUnmount() {
-    window.removeEventListener('click', this.outsideClick)
-  }
+  useEffect(() => {
+    const outsideClick = (event: any) => {
+      if (multiselectRef.current && !multiselectRef.current.contains(event.target)) setOpen(false)
+    }
+    window.addEventListener('click', outsideClick)
+    return () => {
+      window.removeEventListener('click', outsideClick)
+    }
+  }, [])
 
-  toggleOpen() {
-    this.setState((prevState: any) => ({ open: !prevState.open }))
-  }
-
-  localizeOption(option: any) {
-    return (this.props as any).i18n.t(`${(this.props as any).localizationPrefix}.${option}`)
-  }
+  if (!values) return null
 
-  render() {
-    const { values = [] } = this.props
-    if (!values) return null
-    return (
-      <div
-        ref="multiSelect"
-        tabIndex={0}
-        onMouseDown={this.toggleOpen.bind(this)}
-        onFocus={() => this.setState({ open: true })}
-        onBlur={() => this.setState({ open: false })}
-        className={`multi-select ${this.state.open ? 'has-focus' : ''}`}
-        disabled={this.props.disabled}
-      >
-        <div className="multi-select__closed-content">
-          {R.isEmpty(values) ? (
-            <span className="multi-select__placeholder">{(this.props as any).i18n.t('multiSelect.placeholder')}</span>
-          ) : (
-            R.pipe(
-              R.map((option: any) => this.localizeOption(option)),
-              R.join(', ')
-            )(values)
-          )}
-        </div>
-        <div className="multi-select__opened-content-anchor">
-          {this.state.open ? (
-            <div className="multi-select__opened">
-              {R.map(
-                (option: any) => (
-                  <div
-                    className="multi-select__opened-item"
-                    key={option}
-                    onMouseDown={(e) => e.stopPropagation()}
-                    onClick={optionClick(values, (this.props as any).onChange, option)}
-                  >
-                    <span className={`fra-checkbox ${R.contains(option, values) ? 'checked' : ''}`} />
-                    <span className="multi-select__opened-item-label">{this.localizeOption(option)}</span>
-                  </div>
-                ),
-                (this.props as any).options
-              )}
-            </div>
-          ) : null}
-        </div>
+  // @ts-ignore
+  // @ts-ignore
+  return (
+    <div
```

Comment:
maybe it can even become a button ?

Reply:
Should be ok, semantics would change a bit (component left in refactorable components list)
Left out of this PR for now